src/internal_telemetry/allocations/allocator/tracing_allocator.rs

```diff
@@ -48,40 +48,46 @@ impl<A: GlobalAlloc, T: Tracer> GroupedTraceableAllocator<A, T> {
 unsafe impl<A: GlobalAlloc, T: Tracer> GlobalAlloc for GroupedTraceableAllocator<A, T> {
     #[inline]
     unsafe fn alloc(&self, object_layout: Layout) -> *mut u8 {
-        let (group_id_ptr, object_ptr, _wrapped_layout) =
-            self.get_wrapped_allocation(object_layout);
-        let object_size = object_layout.size();
-        // Group id value of zero implies allocations tracking was disabled
-        // during this allocation. We override this if allocations were in fact enabled.
-        group_id_ptr.write(0);
         if TRACK_ALLOCATIONS.load(Ordering::Relaxed) {
+            let (group_id_ptr, object_ptr, _wrapped_layout) =
+                self.get_wrapped_allocation(object_layout);
+            let object_size = object_layout.size();
+            // Group id value of zero implies allocations tracking was disabled
+            // during this allocation. We override this if allocations were in fact enabled.
```

Comment:
This comment, and the one in `dealloc` that got removed around the `if group_id == 0` line, are now out-of-date. If we're here at this line, setting the group ID to zero, we are defacto tracking allocations, even if it's technically only startup.